marklogic-client-api/src/main/java/com/marklogic/client/datamovement/impl/QueryBatcherImpl.java

```diff
@@ -91,6 +91,8 @@
   private Calendar jobEndTime;
   private long maxUris = Long.MAX_VALUE;
   private AtomicBoolean jobDone = new AtomicBoolean(false);
+  private long maxBatches = Long.MAX_VALUE;
+  private boolean isMaxBatchesSet = false;
```

Comment:
Instead of a boolean field, could we test maxBatches < Long.MAX_Value as the guard on setMaxUris()?